### PR TITLE
Barnes & Noble

### LINKED ISSUE #580: Barnes and Noble
https://stores.barnesandnoble.com/stores?lng=-97.7429885864258&lat=30.2675895690918&searchText=Austin%2c+TX&storeFilter=all&v=1&view=map

I didn't assign myself on purpose this time.

### PR DESCRIPTION
**Pull request by MagnetonIO**
Fixes #580

### CODE REVIEW

File: locations/spiders/barnesandnoble.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            opening_hours.append(day_hours.strip())
+
+        return '; '.join(opening_hours)
+
+    def parse(self, response):
+        content = response.body_as_unicode()
+
+        try:
+            stores = json.loads(re.search('var storesJson = (.*)\;', content).group(1))
+        except:
```
Comment (iandees) [inline locations/spiders/barnesandnoble.py]: The try/except here is redundant. Please remove it and let the exception bubble up without catching it.

File: locations/spiders/barnesandnoble.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            raise
+
+        for store in stores:
+            props = {}
+            props['ref'] = store.get('storeId')
+            props['city'] = store.get('city')
+            props['state'] = store.get('state')
+            props['lat'] = store.get('location')[0]
```
Comment (iandees) [inline locations/spiders/barnesandnoble.py]: You have lat and lon flipped here.

**Comment by evanmosseri** [comment]
There are 781 barnes and nobles out there

**Comment by MagnetonIO** [comment]
This spider requires a zip code.

**Comment by MagnetonIO** [comment]
@evanmosseri , this spider needs  zipcode. Now 60001 was hard coded already.

**Comment by iandees** [comment]
I'm going to close this until we find a way around zipcode-only querying.

**Pull request closed without merging**